Add general sections for commands that require/do not require parser and update logic section to work with linkytime (#197)
AY2122S2-CS2103T-T13-3/tp: docs/DeveloperGuide.md

<img src="images/LogicClassDiagram.png" width="550"/>

How the `Logic` component works:
1. When `Logic` is called upon to execute a command, it uses the `LinkyTimeParser` class to parse the user command.
2. This results in a `Command` object (more precisely, an object of one of its subclasses e.g., `AddMeetingCommand`) which is executed by the `LogicManager`.

> 3. The command can communicate with the `Model` when it is executed (e.g. to add a person).

Reviewer: ```suggestion
3. The command can communicate with the `Model` when it is executed (e.g. to add a meeting).
```

Minor typo

of the program and after each command execution.
* Cons: Implementation would sprawl across different components and more effort is required to ensure correctness.

### Commands
This section explains the general implementation of all commands.

> The implementation of all commands can be split into 2 general implementation flow, commands with a parser and commands without.

Reviewer: ```suggestion
The implementation of all commands in LinkyTime can be split into two general implementation flows: commands with a parameter parser, and commands without.
```

Not sure what is the best way to explain that `commands with a parser` means that the parser is command-specific, i.e. they have `XYZCommandParser`

#### Commands with a parser
This section explains the general implementation of all commands that requires a parser to handle additional user input.

Below is the sequence diagram for the execution of these commands (denoted by `XYZCommand`) after user input is sent to `LogicManager`. The execution of each command has been omitted due to their differences and will be covered in the respective command sections.

![`CommandsWithParser` sequence diagram](images/CommandsWithParserSequenceDiagram.png)

Step 1:

> The user enters a command with additional params(requires a parser) which is then passed to the `LogicManager`.

Reviewer: ```suggestion
The user enters a command with additional parameters (which requires a parser) which is then passed to the `LogicManager`.
```

Step 2:
The `LogicManager` then calls `LinkyTimeParser::parseCommand` for it to figure out what command this is.

Step 3:
The `LinkyTimeParser` parses the user input and creates a command parser for that specific command. (denoted by `XYZCommandParser`)

Step 4:

> The parser is then returned to the `LinkyTimeParser` which then calls `XYZCommandParser::parse` to parse the additional params.

Reviewer: ```suggestion
The command parser is then returned to the `LinkyTimeParser` which then calls `XYZCommandParser::parse` to parse the additional parameters.
```

Step 5:
The `XYZCommandParser` then creates its respective command (denoted by `XYZCommand`) and returns it to `LogicManager`.

Step 6:
The `LogicManager` then calls `XYZCommand::execute` where the interaction between the command and the model is handled.

Step 7:
The `XYZCommand` then creates a successful `CommandResult` and returns it to the UI.

#### Commands without a parser
This section explains the general implementation of all commands that does not require a parser.

Below is the sequence diagram for the execution of these commands (denoted by `XYZCommand`) after user input is sent to `LogicManager`. The execution of each command has been omitted due to their differences and will be covered in the respective command sections.

> ![`CommandsWithParser` sequence diagram](images/CommandsWithoutParserSequenceDiagram.png)

Reviewer: ```suggestion
![`CommandsWithoutParser` sequence diagram](images/CommandsWithoutParserSequenceDiagram.png)
```

Step 3:
The `LinkyTimeParser` parses the user input and creates the respective command object (denoted by `XYZCommand`).

Step 4:
The `XYZCommand` is then returned to the `LogicManager`.

Step 5:
The `LogicManager` then calls `XYZCommand::execute` where the interaction between the command and the model is handled.

Step 6:
The `XYZCommand` then creates a successful `CommandResult` and returns it to the UI.

#### List Meetings feature
This section explains the implementation of the List Meetings feature via the `list` command.
The `ListMeetingCommand` updates the UI to display the details of all upcoming meetings in `LinkyTime`.
It is a command that [does not require a parser](#Commands without a parser). 

Below is the sequence diagram reference frame for the execution of `ListMeetingCommand`.

![`ListMeetingCommand` sequence diagram](images/ListMeetingSequenceDiagramReferenceFrame.png)

Step 1:
The `LogicManager` calls `ListMeetingCommand::execute` with the returned `ListMeetingCommand`.

Step 2:
The `ListMeetingCommand` then calls `Model::showCompletedMeetings` to update the meeting list to show only upcoming meetings.

Step 3:

> The `ListMeetingCommand` then continues its execution as defined by [this](#Commands without a parser) sequence diagram.

Reviewer 2: I think the hyperlink got problem
